File: SDK/CentralService/java/client/src/main/java/erp/centralservice/client/errors/CentralServiceErrorKind.java
package erp.centralservice.client.errors;

/**
 * 中心服务错误响应的已知分类。
 */
public enum CentralServiceErrorKind {
    /** 所有候选端点都因传输层失败或熔断跳过而不可用。 */
    Transport,
    /** 命中了 SDK 约定的统一 {@code ApiResponse} 错误结构。 */
    ApiResponse,
    /** 命中了 ASP.NET Core {@code ProblemDetails} 结构。 */
    ProblemDetails,
    /** 命中了带有 {@code errors} 节点的验证错误结构。 */
    ValidationProblemDetails,
    /** 响应体不是 JSON，或 JSON 解析失败后退回纯文本解释。 */
    PlainText,
    /** 无法识别响应结构时使用的兜底分类。 */
    Unknown
}

File: SDK/CentralService/dotnet/src/CentralService.Client/Errors/CentralServiceErrorKind.cs
namespace CentralService.Client.Errors
{
    /// <summary>
    /// 表示中心服务错误响应的解析类型。
    /// </summary>
    public enum CentralServiceErrorKind
    {
        /// <summary>
        /// 服务端返回了统一的 API 响应错误结构。
        /// </summary>
        ApiResponse = 1,

        /// <summary>
        /// 服务端返回了 ProblemDetails 结构。
        /// </summary>
        ProblemDetails = 2,

        /// <summary>
        /// 服务端返回了包含字段校验信息的 ValidationProblemDetails 结构。
        /// </summary>
        ValidationProblemDetails = 3,

        /// <summary>
        /// 服务端返回了纯文本错误内容。
        /// </summary>
        PlainText = 4,

        /// <summary>
        /// 错误响应无法归类到已知结构。
        /// </summary>
        Unknown = 5,

        /// <summary>
        /// 所有候选端点都因传输异常或熔断跳过而无法完成请求。
        /// </summary>
        Transport = 6
    }
}
